src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,158 @@
+     private DoubleEntry m_armDSubs;
+     private DoubleEntry m_armIzSubs;
+     private DoubleEntry m_armFFSubs;
+     private DoublePublisher m_armSetpointPub;   
+     private DoublePublisher m_armVelPub;
+     private DoublePublisher m_armFFTestingVolts;
+     private DoubleEntry m_armOffset;
+     private DoublePublisher m_targetAngle;
+     private DoublePublisher m_armPosPub;
+
+    //for  arm ff
+    private DoubleEntry m_armMomentToVoltage;
+
+    //spark absolute encoder
+    SparkAbsoluteEncoder m_absEncoder;  
+    //embedded relative encoder
+    //private RelativeEncoder m_Encoder;
+    public SparkPIDController m_pidControllerArm;  
+
+    ProfiledPIDFFController m_profiledFFController = new ProfiledPIDFFController();
+
+    public static ArmSubsystem getInstance() {
+      if (instance == null) {
+        SubsystemChecker.subsystemConstructed(SubsystemType.ArmSubsystem);
+        instance = new ArmSubsystem();
+      }
+      return instance;
+    }
+    public ArmSubsystem() {
+      m_arm = new CANSparkMax(Config.CANID.ARM_SPARK_CAN_ID, motorType); //creates SparkMax motor controller 
+      ErrorCheck.errREV(m_arm.restoreFactoryDefaults());
+      ErrorCheck.errREV(m_arm.setSmartCurrentLimit(Config.ArmConfig.CURRENT_LIMIT));
+      m_arm.setInverted(Config.ArmConfig.SET_INVERTED); //sets movement direction
+      ErrorCheck.errREV(m_arm.setIdleMode(IdleMode.kBrake)); //sets brakes when there is no motion
+      
+      ErrorCheck.errREV(m_arm.setSoftLimit(SoftLimitDirection.kForward, Config.ArmConfig.arm_forward_limit));
+      ErrorCheck.errREV(m_arm.setSoftLimit(SoftLimitDirection.kReverse, Config.ArmConfig.arm_reverse_limit));
+      ErrorCheck.errREV(m_arm.enableSoftLimit(SoftLimitDirection.kForward, Config.ArmConfig.SOFT_LIMIT_ENABLE));
+      ErrorCheck.errREV(m_arm.enableSoftLimit(SoftLimitDirection.kReverse, Config.ArmConfig.SOFT_LIMIT_ENABLE));
+
+      //SparkMax periodic status frame 5: frequency absolute encoder position data is sent over the can bus
+      //SparkMax perioidc status frame 6: frequency absolute encoder velocity data is sent over the can bus
+      m_arm.setPeriodicFramePeriod(PeriodicFrame.kStatus5, 20);
+      m_arm.setPeriodicFramePeriod(PeriodicFrame.kStatus6, 20);
+
+      m_absEncoder = m_arm.getAbsoluteEncoder(Type.kDutyCycle);
+      m_absEncoder.setInverted(true);
+      m_absEncoder.setPositionConversionFactor(Config.ArmConfig.armPositionConversionFactor);
+      m_absEncoder.setVelocityConversionFactor(Config.ArmConfig.armVelocityConversionFactor);
+      m_absEncoder.setZeroOffset(Math.toRadians(Config.ArmConfig.armAbsEncoderOffset));
+
+      m_pidControllerArm = m_arm.getPIDController();
+      m_pidControllerArm.setFeedbackDevice(m_absEncoder);
+
+      
+    NetworkTable ArmTuningTable = NetworkTableInstance.getDefault().getTable(m_tuningTable);
+    m_armPSubs = ArmTuningTable.getDoubleTopic("P").getEntry(Config.ArmConfig.arm_kP, PubSubOption.periodic(0.02));
```

Comment:
This doesn't need to be on a faster periodic rate. You can remove the `PubSubOption.periodic(0.02)` part.